src/test/librados/io_cxx.cc

```diff
@@ -218,6 +218,38 @@ TEST_F(LibRadosIoPP, SparseReadOpPP) {
   }
 }
 
+TEST_F(LibRadosIoPP, SparseReadExtentArrayOpPP) {
+  int buf_len = 128;
+  char buf[buf_len], zbuf[buf_len];
+  memset(buf, 0xcc, buf_len);
+  memset(zbuf, 0, buf_len);
+  bufferlist bl;
+  int i, len = 1024, skip = 5;
+  bl.append(buf, buf_len);
+  for (i = 0; i < len; i += skip) {
+    ASSERT_EQ(0, ioctx.write("sparse-read", bl, bl.length(), i * buf_len));
+  }
+  ASSERT_EQ(0, ioctx.write("sparse-read", bl, bl.length(), (len - 1) * buf_len));
```

Comment:
Nit: now that you have changed which blocks are operated on, the previous version is better -- it matches what `expect_bl` loop is doing:
```suggestion
  for (i = 0; i < len; i++) {
    if (!(i % skip) || i == (len - 1)) {
      ASSERT_EQ(0, ioctx.write("sparse-read", bl, bl.length(), i * buf_len));
    }
  }
```

Reply:
Done, fixed it.